src/util/address.rs

```diff
@@ -727,6 +727,26 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Determine if an address is related to a given secp256k1 public key
+    pub fn is_related_to_secp256k1_key(
+        &self,
+        pubkey: &PublicKey 
+    ) -> Result<bool, Error> {
+        let address_data = match bech32_decode(self) {
```

Comment:
This could be easier to read and understand using an `if let` clause:

eg:
```rust
if let Ok(data) = bech32_decode(self) {
    Ok(*pubkey.pubkey_hash() == *data)
} else {
    let pubkey_hash = pubkey.pubkey_hash();
    let redeem_hash = segwit_redeem_hash(&pubkey_hash);
    let base58_check = get_payload_bytes(self);
    Ok(
        (*pubkey_hash == *base58_check) ||
        (*redeem_hash == *base58_check)
    )
}
```

Reply:
I like the 'if let', however this function never returns an error so could be simplified even further to:
```
    pub fn is_related_to_key(&self, pubkey: &PublicKey) -> bool {
        if let Ok(data) = bech32_decode(self) {
            return *pubkey.pubkey_hash() == *data;
        }
        
        let pubkey_hash = pubkey.pubkey_hash();
        let redeem_hash = segwit_redeem_hash(&pubkey_hash);
        let base58_check = get_payload_bytes(self);
        *pubkey_hash == *base58_check || *redeem_hash == *base58_check
    }
```

Reply:
Thank you for the suggestion. Updated in a046687857

Reply:
Note, the parameter is _not_ a secp256k1 pubkey so perhaps we can better describe the function by dropping the 'secp256k1' from the function name.

Reply:
I was conflicted here (as noted in description). I wanted to make it clear that this is not a schnorr key. After all PublicKey is effectively a newtype around `secp256k1::PublicKey`

```rust
pub struct PublicKey {
    /// Whether this public key should be serialized as compressed
    pub compressed: bool,
    /// The actual ECDSA key
    pub inner: secp256k1::PublicKey,
}
```